functional-tests/test-bed/src/main/java/org/arquillian/smart/testing/ftest/testbed/project/ProjectBuilder.java

```diff
@@ -5,35 +5,109 @@
 
-    ProjectBuilder(Path root) {
+    private int remotePort = DEFAULT_DEBUG_PORT;
+    private boolean quietMode = true;
+    private boolean enableRemoteDebugging = false;
+    private boolean suspend = true;
+    private boolean mvnDebugOutput;
+
+    ProjectBuilder(Path root, Project project) {
         this.root = root;
+        this.project = project;
+    }
+
+    public Project configure() {
+        return this.project;
+    }
+
+    /**
+     * Enables remote debugging of embedded maven build so we can troubleshoot our extension and provider
+     * By default it sets suspend to 'y', so the build will wait until we attach remote debugger to the port DEFAULT_DEBUG_PORT
+     */
+    public ProjectBuilder withRemoteDebugging() {
+        return withRemoteDebugging(DEFAULT_DEBUG_PORT, true);
+    }
+
+    public ProjectBuilder withRemoteDebugging(int port) {
+        return withRemoteDebugging(port, true);
+    }
+
+    public ProjectBuilder withRemoteDebugging(int port, boolean suspend) {
+        this.enableRemoteDebugging = true;
+        this.remotePort = port;
+        this.suspend = suspend;
+        return this;
+    }
+
+    public ProjectBuilder quiet(boolean quiet) {
+        this.quietMode = quiet;
+        return this;
+    }
+
+    /**
+     * Enables mvn debug output (-X) flag.
+     */
+    public ProjectBuilder withDebugOutput() {
+        this.mvnDebugOutput = true;
+        quiet(false);
+        return this;
+    }
+
+    public ProjectBuilder withSystemProperties(String ... systemPropertiesPairs) {
+        if (systemPropertiesPairs.length % 2 != 0) {
+            throw new IllegalArgumentException("Expecting even amount of variable name - value pairs to be passed. Got "
+                + systemPropertiesPairs.length
+                + " entries. "
+                + Arrays.toString(systemPropertiesPairs));
+        }
+
+        for (int i = 0; i < systemPropertiesPairs.length; i += 2) {
+            this.systemProperties.put(systemPropertiesPairs[i], systemPropertiesPairs[i + 1]);
+        }
+
+        return this;
     }
 
     List<TestResult> build(String... goals) {
-        final BuiltProject build = EmbeddedMaven.forProject(root.toAbsolutePath().toString() + "/pom.xml")
+        final PomEquippedEmbeddedMaven embeddedMaven =
+            EmbeddedMaven.forProject(root.toAbsolutePath().toString() + "/pom.xml");
+
+        if (isRemoteDebugEnabled()) {
+            final String debugOptions = String.format(MVN_DEBUG_MODE, shouldSuspend(), getRemotePort());
+            System.out.println(">>> Executing build with debug options: " + debugOptions);
+            embeddedMaven.setMavenOpts(debugOptions);
+        }
+
+        final BuiltProject build = embeddedMaven
                     .setGoals(goals)
-                    .setQuiet()
+                    .setDebug(isMavenDebugOutputEnabled())
+                    .setQuiet(!isMavenDebugOutputEnabled() && quietMode)
```

Comment:
For me are two different things, one is that you want to debug and the other one is that you want to print things, tester should enable both if he wants to see both. But it is just my opinion and any other way is fine for me.